proto/scheduler.proto

```diff
@@ -122,6 +122,26 @@ message SchedulingMetadata {
   string executor_group_id = 5;
   // Group ID of the user that issued the Execute request.
   string task_group_id = 6;
+
+  // Firecracker remote snapshot sharing options. This controls which snapshot
+  // the task prefers to use, as well as which snapshot it will update in the
+  // remote cache, if any. If not set, snapshots will only be shared locally,
+  // and will use the platform hash as the routing key.
+  RemoteSnapshotOptions remote_snapshot_options = 7;
```

Comment:
I originally considered putting these in platform props but then realized that the platform hash is part of the snapshot key and also messes with the task router, so it won't quite work without some ugly hacks.

I figure it will be cleaner to have this as scheduling metadata, which we can pass to the scheduler as a serialized proto via gRPC header in our /Execute req, then ultimately access it in the executor via the ExecutionTask.

wdyt @tylerwilliams @maggie-lou ?